lib/auth/grpcserver.go

```diff
@@ -840,6 +869,18 @@ func (g *GRPCServer) SetAccessRequestState(ctx context.Context, req *authpb.Requ
 }
 
 func (g *GRPCServer) SubmitAccessReview(ctx context.Context, review *types.AccessReviewSubmission) (*types.AccessRequestV3, error) {
+	// Prevent users from submitting access reviews with the "promoted" state.
+	// Promotion is only allowed by SubmitAccessReviewAllowPromotion API in Ent.
+	if review.Review.ProposedState.IsPromoted() {
+		return nil, trace.BadParameter("state promoted can be only set when promoting to access list")
+	}
+
+	return g.SubmitAccessReviewAllowPromotion(ctx, review)
```

Comment:
this was confusing to me (b/c we rejected it above), i would rather duplicate a few lines of code and keep it how it was before?

Reply:
I'd be very happy to do that, but `SubmitAccessReviewAllowPromotion` was only created as an entry point for the promotion request and it's called from `e`. Otherwise, I've got no idea how can we call the logic in OSS.